packages/gateway-client/src/worker/p2p-client/stream-factory.ts

```diff
@@ -5,16 +5,110 @@ import type { P2pClient } from '.';
 import { ServerPeerStatus } from '../../worker-messaging';
 import { logger } from '../logging';
 import status from '../status';
+import { decode } from '../p2p-fetch/lob-enc';
+import { Deferred } from '../p2p-fetch/p2p-request';
 
 const waitFor = async (t: number): Promise<void> =>
     new Promise(r => setTimeout(r, t));
 
+export class RequestStream {
+    private log = logger.getLogger('worker/p2p/wrapped-stream');
+    private outbox = new Deferred<Buffer[]>();
+    private inbox = new Deferred<Buffer[]>();
+    private hasOpened = false;
+
+    get isOpen(): boolean {
+        return this.stream.stat.timeline.close === undefined;
+    }
+
+    constructor(private readonly stream: Stream) {}
+
+    // set up stream for continual send/receive
+    public async open(): Promise<void> {
+        this.hasOpened = true;
+        // we can only call sink once, so we need to provide a generator
+        // that will yield chunks from the outbox
+        // eslint-disable-next-line @typescript-eslint/no-this-alias
+        const that = this;
+        this.stream.sink(
+            (async function* () {
+                while (true) {
+                    const chunks = await that.outbox.promise;
+
+                    for await (const chunk of chunks) {
+                        yield chunk;
```

Comment:
Just wanted to highlight that you're now yielding a `Buffer` instead of a `UInt8Array`

Reply:
yeah, that's intentional, and Buffer inherits from Uint8Array so anything that depends on that api will still work.